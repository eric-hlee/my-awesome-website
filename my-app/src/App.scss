@import "~bootstrap/scss/bootstrap";

$cultured: #f9f8f8;
$home: url("assets/images/home.jpg");

header,
section,
main {
	padding: 0.5rem 1rem;
}

.section {
	position: relative;
}

a {
	color: inherit;
}

.content {
	min-height: calc(100vh - 70px);
}

.footer {
	background-color: #f8f9fa !important;
	font-size: 14px;
	height: 100px;
	padding-top: 20px;
	text-align: center;
	width: 100%;
}

.dot {
	height: 25px;
	width: 25px;
	background-color: #0e76a8;
	border-radius: 50%;
	display: inline-block;
}

.key-skills {
	.list-group-item {
		max-width: 150px;
	}
}

.nav-link {
	&.active {
		font-weight: bold;
	}
}

.underline {
	text-decoration: underline;
}

#info-item {
	padding: 4px;
}

.carousel-indicators {
	bottom: -50px;
	li {
		background-color: gray;
	}
}

.card {
	border-top: 0;
	border-top-left-radius: 0;
	border-top-right-radius: 0;
}

.card-img-top {
	width: 20rem;
}

.jumbotron {
	background-image: $home;
	background-position: center;
	background-size: cover;
}

.img-eric {
	max-width: fit-content;
}

.speech-bubble {
	background: #efefef;
	border-radius: 4px;
	margin: 0 auto 40px 0;
	padding: 15px;
	position: relative;
	p {
		font-size: 1.2rem;
		margin: 0 0 10px;
		:last-of-type {
			margin-bottom: 0;
		}
	}
	&::after {
		border-left: 20px solid transparent;
		border-top: 20px solid #efefef;
		content: "";
		left: -20px;
		top: 20px;
		position: absolute;
	}
}
