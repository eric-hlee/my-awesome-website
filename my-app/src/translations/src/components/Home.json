[{"id":"home.greeting","defaultMessage":"Hello!","file":"src/components/Home.js","start":{"line":18,"column":7},"end":{"line":21,"column":9}},{"id":"home.whoami","defaultMessage":"My name is Eric.","file":"src/components/Home.js","start":{"line":24,"column":7},"end":{"line":27,"column":9}},{"id":"home.resume","defaultMessage":"Resume","file":"src/components/Home.js","start":{"line":35,"column":8},"end":{"line":38,"column":10}},{"id":"home.life_as_dev","defaultMessage":"🧑🏻‍💻 My life as a developer:","file":"src/components/Home.js","start":{"line":59,"column":11},"end":{"line":62,"column":13}},{"id":"home.life_as_dev_1","defaultMessage":"Started learning HTML at 10","file":"src/components/Home.js","start":{"line":67,"column":11},"end":{"line":70,"column":13}},{"id":"home.life_as_dev_2","defaultMessage":"Received an award for creating a website at 12","file":"src/components/Home.js","start":{"line":73,"column":11},"end":{"line":76,"column":13}},{"id":"home.life_as_dev_3","defaultMessage":"Studied computer science in university at 18","file":"src/components/Home.js","start":{"line":79,"column":11},"end":{"line":82,"column":13}},{"id":"home.life_as_dev_4","defaultMessage":"Completed an internship at 22","file":"src/components/Home.js","start":{"line":85,"column":11},"end":{"line":88,"column":13}},{"id":"home.life_as_dev_5","defaultMessage":"Kicked off career as a technical support engineer at 23","file":"src/components/Home.js","start":{"line":91,"column":11},"end":{"line":94,"column":13}},{"id":"home.life_as_dev_6","defaultMessage":"Joined a startup as a junior software engineer at 24","file":"src/components/Home.js","start":{"line":97,"column":11},"end":{"line":100,"column":13}},{"id":"home.life_as_dev_7","defaultMessage":"Got laid off due to Covid-19 and looking for new opportunities!","file":"src/components/Home.js","start":{"line":103,"column":11},"end":{"line":106,"column":13}},{"id":"home.learn_more","defaultMessage":"Learn more about me","file":"src/components/Home.js","start":{"line":119,"column":12},"end":{"line":122,"column":14}}]
