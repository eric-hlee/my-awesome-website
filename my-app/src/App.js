import React from "react";
import { Switch, Route } from "react-router-dom";
import About from "./components/About";
import Home from "./components/Home";
import NavBar from "./components/NavBar";
import "./App.scss";
import "./assets/fontello/css/fontello.css";

export default function App() {
  return (
    <div className="content">
      <NavBar />
      <Switch>
        <Route path="/about">
          <About />
        </Route>
        <Route path="/">
          <Home />
        </Route>
      </Switch>
    </div>
  );
}
