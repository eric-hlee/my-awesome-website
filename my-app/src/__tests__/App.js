import React from "react";
import { shallow, mount } from "enzyme";
import { MemoryRouter } from "react-router";
import App from "../App";
import Home from "../components/Home";
import About from "../components/About";

describe("<App />", () => {
	it("renders IntlProvider, NavBar, BrowserRouter, Switch and Route", () => {
		const wrapper = shallow(<App />);
		expect(wrapper).toMatchSnapshot();
	});

	it("renders Home", () => {
		const wrapper = mount(
			<MemoryRouter>
				<App />
			</MemoryRouter>
		);
		expect(wrapper).toMatchSnapshot();
		expect(wrapper.find(Home)).toHaveLength(1);
		expect(
			wrapper.find("Router").first().prop("history").location.pathname
		).toEqual("/");
	});

	it("renders About", () => {
		const wrapper = mount(
			<MemoryRouter initialEntries={["/about"]}>
				<App />
			</MemoryRouter>
		);
		expect(
			wrapper.find("Router").first().prop("history").location.pathname
		).toEqual("/about");
	});
});
