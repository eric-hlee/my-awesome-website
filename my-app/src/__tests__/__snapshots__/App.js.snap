// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> renders Home 1`] = `
<MemoryRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "canGo": [Function],
        "createHref": [Function],
        "entries": Array [
          Object {
            "hash": "",
            "key": "5j6yzc",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
        ],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "index": 0,
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "key": "5j6yzc",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <App>
      <div
        className="content"
      >
        <IntlProvider
          defaultFormats={Object {}}
          defaultLocale="en"
          formats={Object {}}
          locale="en"
          messages={
            Object {
              "about.education": "🎓 Education",
              "about.key_skills": "🧰 Key Skills",
              "about.work_experience": "📃 Work Experience",
              "education.deans_list": "Dean’s Honour List 2017-18",
              "education.degree": "Bachelor of Computing (Honours)",
              "education.program": "Specialization in Software Design with Professional Internship",
              "education.queensu": "Queen's University",
              "home.greeting": "Hello 👋",
              "home.learn_more": "Learn more about me",
              "home.life_as_dev": "🧑🏻‍💻 My life as a developer:",
              "home.life_as_dev_1": "Started learning HTML at 10",
              "home.life_as_dev_2": "Received an award for creating a website at 12",
              "home.life_as_dev_3": "Studied computer science in university at 18",
              "home.life_as_dev_4": "Completed an internship at 22",
              "home.life_as_dev_5": "Kicked off career as a technical support engineer at 23",
              "home.life_as_dev_6": "Joined a startup as a junior software engineer at 24",
              "home.life_as_dev_7": "🆕Open for new opportunities!",
              "home.resume": "Resume",
              "home.whoami": "My name is Eric.",
              "navbar.about": "About",
              "navbar.home": "Home",
              "navbar.text": "Eric Lee",
              "work.conv": "Conversocial",
              "work.conv_jse": "Junior Software Engineer",
              "work.conv_work_period": "May 2019 – Apr 2020",
              "work.ibm": "IBM",
              "work.ibm_intern": "e-Commerce Java Development Support Intern",
              "work.ibm_work_period": "May 2016 – Apr 2018",
              "work.kingston_on": "Kingston, Ontario",
              "work.london_uk": "London, United Kingdom",
              "work.queensu": "Queen's University",
              "work.queensu_ta": "Teaching Assistant",
              "work.queensu_work_period": "Sep 2017 – Apr 2018",
              "work.toronto_ca": "Toronto, Canada",
              "work.verv": "Verv",
              "work.verv_tse": "Technical Support Engineer",
              "work.verv_work_period": "Oct 2018 – Apr 2019",
            }
          }
          onError={[Function]}
          textComponent={Symbol(react.fragment)}
        >
          <NavBar>
            <Navbar
              bg="light"
              collapseOnSelect={false}
              expand="lg"
              variant="light"
            >
              <nav
                className="navbar navbar-expand-lg navbar-light bg-light"
              >
                <NavbarBrand
                  href="/"
                >
                  <a
                    className="navbar-brand"
                    href="/"
                  >
                    <FormattedMessage
                      defaultMessage="Eric Lee"
                      id="navbar.text"
                      values={Object {}}
                    >
                      Eric Lee
                    </FormattedMessage>
                  </a>
                </NavbarBrand>
                <NavbarToggle
                  aria-controls="basic-navbar-nav"
                  label="Toggle navigation"
                >
                  <button
                    aria-controls="basic-navbar-nav"
                    aria-label="Toggle navigation"
                    className="navbar-toggler collapsed"
                    onClick={[Function]}
                    type="button"
                  >
                    <span
                      className="navbar-toggler-icon"
                    />
                  </button>
                </NavbarToggle>
                <NavbarCollapse
                  id="basic-navbar-nav"
                >
                  <Collapse
                    appear={false}
                    dimension="height"
                    getDimensionValue={[Function]}
                    id="basic-navbar-nav"
                    in={false}
                    mountOnEnter={false}
                    timeout={300}
                    unmountOnExit={false}
                  >
                    <Transition
                      addEndListener={[Function]}
                      appear={false}
                      aria-expanded={null}
                      enter={true}
                      exit={true}
                      id="basic-navbar-nav"
                      in={false}
                      mountOnEnter={false}
                      onEnter={[Function]}
                      onEntered={[Function]}
                      onEntering={[Function]}
                      onExit={[Function]}
                      onExited={[Function]}
                      onExiting={[Function]}
                      timeout={300}
                      unmountOnExit={false}
                    >
                      <div
                        aria-expanded={null}
                        className="navbar-collapse collapse"
                        id="basic-navbar-nav"
                      >
                        <Nav
                          className="mr-auto"
                          fill={false}
                          justify={false}
                        >
                          <ForwardRef
                            as="div"
                            className="mr-auto navbar-nav"
                            onSelect={[Function]}
                          >
                            <div
                              className="mr-auto navbar-nav"
                              onKeyDown={[Function]}
                            >
                              <NavLink
                                as={
                                  Object {
                                    "$$typeof": Symbol(react.forward_ref),
                                    "displayName": "SafeAnchor",
                                    "render": [Function],
                                  }
                                }
                                disabled={false}
                                href="/"
                              >
                                <ForwardRef
                                  as={
                                    Object {
                                      "$$typeof": Symbol(react.forward_ref),
                                      "displayName": "SafeAnchor",
                                      "render": [Function],
                                    }
                                  }
                                  className="nav-link"
                                  disabled={false}
                                  href="/"
                                >
                                  <SafeAnchor
                                    className="nav-link"
                                    data-rb-event-key="/"
                                    disabled={false}
                                    href="/"
                                    onClick={[Function]}
                                  >
                                    <a
                                      className="nav-link"
                                      data-rb-event-key="/"
                                      href="/"
                                      onClick={[Function]}
                                      onKeyDown={[Function]}
                                    >
                                      <FormattedMessage
                                        defaultMessage="Home"
                                        id="navbar.home"
                                        values={Object {}}
                                      >
                                        Home
                                      </FormattedMessage>
                                    </a>
                                  </SafeAnchor>
                                </ForwardRef>
                              </NavLink>
                              <NavLink
                                as={
                                  Object {
                                    "$$typeof": Symbol(react.forward_ref),
                                    "displayName": "SafeAnchor",
                                    "render": [Function],
                                  }
                                }
                                className="about"
                                disabled={false}
                                href="/about"
                              >
                                <ForwardRef
                                  as={
                                    Object {
                                      "$$typeof": Symbol(react.forward_ref),
                                      "displayName": "SafeAnchor",
                                      "render": [Function],
                                    }
                                  }
                                  className="about nav-link"
                                  disabled={false}
                                  href="/about"
                                >
                                  <SafeAnchor
                                    className="about nav-link"
                                    data-rb-event-key="/about"
                                    disabled={false}
                                    href="/about"
                                    onClick={[Function]}
                                  >
                                    <a
                                      className="about nav-link"
                                      data-rb-event-key="/about"
                                      href="/about"
                                      onClick={[Function]}
                                      onKeyDown={[Function]}
                                    >
                                      <FormattedMessage
                                        defaultMessage="About"
                                        id="navbar.about"
                                        values={Object {}}
                                      >
                                        About
                                      </FormattedMessage>
                                    </a>
                                  </SafeAnchor>
                                </ForwardRef>
                              </NavLink>
                            </div>
                          </ForwardRef>
                        </Nav>
                      </div>
                    </Transition>
                  </Collapse>
                </NavbarCollapse>
              </nav>
            </Navbar>
          </NavBar>
          <BrowserRouter>
            <Router
              history={
                Object {
                  "action": "POP",
                  "block": [Function],
                  "createHref": [Function],
                  "go": [Function],
                  "goBack": [Function],
                  "goForward": [Function],
                  "length": 1,
                  "listen": [Function],
                  "location": Object {
                    "hash": "",
                    "pathname": "/",
                    "search": "",
                    "state": undefined,
                  },
                  "push": [Function],
                  "replace": [Function],
                }
              }
            >
              <Switch>
                <Route
                  component={[Function]}
                  computedMatch={
                    Object {
                      "isExact": true,
                      "params": Object {},
                      "path": "/",
                      "url": "/",
                    }
                  }
                  exact={true}
                  location={
                    Object {
                      "hash": "",
                      "pathname": "/",
                      "search": "",
                      "state": undefined,
                    }
                  }
                  path="/"
                >
                  <Home
                    history={
                      Object {
                        "action": "POP",
                        "block": [Function],
                        "createHref": [Function],
                        "go": [Function],
                        "goBack": [Function],
                        "goForward": [Function],
                        "length": 1,
                        "listen": [Function],
                        "location": Object {
                          "hash": "",
                          "pathname": "/",
                          "search": "",
                          "state": undefined,
                        },
                        "push": [Function],
                        "replace": [Function],
                      }
                    }
                    location={
                      Object {
                        "hash": "",
                        "pathname": "/",
                        "search": "",
                        "state": undefined,
                      }
                    }
                    match={
                      Object {
                        "isExact": true,
                        "params": Object {},
                        "path": "/",
                        "url": "/",
                      }
                    }
                  >
                    <main>
                      <header>
                        <Jumbotron
                          fluid={false}
                        >
                          <div
                            className="jumbotron"
                          >
                            <h1>
                              <FormattedMessage
                                defaultMessage="Hello!"
                                id="home.greeting"
                                values={Object {}}
                              >
                                Hello 👋
                              </FormattedMessage>
                            </h1>
                            <p
                              className="white"
                            >
                              <FormattedMessage
                                defaultMessage="My name is Eric."
                                id="home.whoami"
                                values={Object {}}
                              >
                                My name is Eric.
                              </FormattedMessage>
                            </p>
                            <p>
                              <Button
                                active={false}
                                className="icon-download"
                                disabled={false}
                                href="http://localhost:9000/resume"
                                type="button"
                                variant="primary"
                              >
                                <SafeAnchor
                                  className="icon-download btn btn-primary"
                                  disabled={false}
                                  href="http://localhost:9000/resume"
                                >
                                  <a
                                    className="icon-download btn btn-primary"
                                    href="http://localhost:9000/resume"
                                    onClick={[Function]}
                                    onKeyDown={[Function]}
                                  >
                                    <FormattedMessage
                                      defaultMessage="Resume"
                                      id="home.resume"
                                      values={Object {}}
                                    >
                                      Resume
                                    </FormattedMessage>
                                  </a>
                                </SafeAnchor>
                              </Button>
                            </p>
                          </div>
                        </Jumbotron>
                      </header>
                      <section>
                        <Container
                          fluid={true}
                        >
                          <div
                            className="container-fluid"
                          >
                            <Row
                              noGutters={false}
                            >
                              <div
                                className="row"
                              >
                                <Col
                                  className="img-eric"
                                >
                                  <div
                                    className="img-eric col"
                                  >
                                    <Image
                                      alt="Picture of Eric"
                                      fluid={true}
                                      height="180"
                                      rounded={false}
                                      roundedCircle={false}
                                      src="test-file-stub"
                                      thumbnail={false}
                                      width="171"
                                    >
                                      <img
                                        alt="Picture of Eric"
                                        className="img-fluid"
                                        height="180"
                                        src="test-file-stub"
                                        width="171"
                                      />
                                    </Image>
                                  </div>
                                </Col>
                                <Col>
                                  <div
                                    className="col"
                                  >
                                    <div
                                      className="speech-bubble"
                                    >
                                      <p>
                                        <strong>
                                          <FormattedMessage
                                            defaultMessage="🧑🏻‍💻 My life as a developer:"
                                            id="home.life_as_dev"
                                            values={Object {}}
                                          >
                                            🧑🏻‍💻 My life as a developer:
                                          </FormattedMessage>
                                        </strong>
                                      </p>
                                      <ul>
                                        <li>
                                          <FormattedMessage
                                            defaultMessage="Started learning HTML at 10"
                                            id="home.life_as_dev_1"
                                            values={Object {}}
                                          >
                                            Started learning HTML at 10
                                          </FormattedMessage>
                                        </li>
                                        <li>
                                          <FormattedMessage
                                            defaultMessage="Received an award for creating a website at 12"
                                            id="home.life_as_dev_2"
                                            values={Object {}}
                                          >
                                            Received an award for creating a website at 12
                                          </FormattedMessage>
                                        </li>
                                        <li>
                                          <FormattedMessage
                                            defaultMessage="Studied computer science in university at 18"
                                            id="home.life_as_dev_3"
                                            values={Object {}}
                                          >
                                            Studied computer science in university at 18
                                          </FormattedMessage>
                                        </li>
                                        <li>
                                          <FormattedMessage
                                            defaultMessage="Completed an internship at 22"
                                            id="home.life_as_dev_4"
                                            values={Object {}}
                                          >
                                            Completed an internship at 22
                                          </FormattedMessage>
                                        </li>
                                        <li>
                                          <FormattedMessage
                                            defaultMessage="Kicked off career as a technical support engineer at 23"
                                            id="home.life_as_dev_5"
                                            values={Object {}}
                                          >
                                            Kicked off career as a technical support engineer at 23
                                          </FormattedMessage>
                                        </li>
                                        <li>
                                          <FormattedMessage
                                            defaultMessage="Joined a startup as a junior software engineer at 24"
                                            id="home.life_as_dev_6"
                                            values={Object {}}
                                          >
                                            Joined a startup as a junior software engineer at 24
                                          </FormattedMessage>
                                        </li>
                                        <li>
                                          <FormattedMessage
                                            defaultMessage="Got laid off due to Covid-19 and looking for new opportunities!"
                                            id="home.life_as_dev_7"
                                            values={Object {}}
                                          >
                                            🆕Open for new opportunities!
                                          </FormattedMessage>
                                        </li>
                                        <li>
                                          <span
                                            aria-label="click on"
                                            role="img"
                                          >
                                            ➡️
                                          </span>
                                          <a
                                            className="underline"
                                            href="/about"
                                          >
                                            <FormattedMessage
                                              defaultMessage="Learn more about me"
                                              id="home.learn_more"
                                              values={Object {}}
                                            >
                                              Learn more about me
                                            </FormattedMessage>
                                          </a>
                                        </li>
                                      </ul>
                                    </div>
                                  </div>
                                </Col>
                              </div>
                            </Row>
                          </div>
                        </Container>
                      </section>
                    </main>
                    <Footer>
                      <footer
                        className="footer"
                      >
                        <p
                          className="icon-copyright"
                        >
                          2020 Eric Lee. Designed and built with
                           
                          <span
                            aria-label="love"
                            role="img"
                          >
                            ❤️
                          </span>
                          .
                        </p>
                        <a
                          href="http://www.linkedin.com/in/erichsuanlee"
                        >
                          <span
                            className="dot icon-linkedin"
                          />
                        </a>
                        <a
                          href="http://github.com/ericlee1995"
                        >
                          <span
                            className="dot icon-github"
                          />
                        </a>
                      </footer>
                    </Footer>
                  </Home>
                </Route>
              </Switch>
            </Router>
          </BrowserRouter>
        </IntlProvider>
      </div>
    </App>
  </Router>
</MemoryRouter>
`;

exports[`<App /> renders IntlProvider, NavBar, BrowserRouter, Switch and Route 1`] = `
<div
  className="content"
>
  <IntlProvider
    defaultFormats={Object {}}
    defaultLocale="en"
    formats={Object {}}
    locale="en"
    messages={
      Object {
        "about.education": "🎓 Education",
        "about.key_skills": "🧰 Key Skills",
        "about.work_experience": "📃 Work Experience",
        "education.deans_list": "Dean’s Honour List 2017-18",
        "education.degree": "Bachelor of Computing (Honours)",
        "education.program": "Specialization in Software Design with Professional Internship",
        "education.queensu": "Queen's University",
        "home.greeting": "Hello 👋",
        "home.learn_more": "Learn more about me",
        "home.life_as_dev": "🧑🏻‍💻 My life as a developer:",
        "home.life_as_dev_1": "Started learning HTML at 10",
        "home.life_as_dev_2": "Received an award for creating a website at 12",
        "home.life_as_dev_3": "Studied computer science in university at 18",
        "home.life_as_dev_4": "Completed an internship at 22",
        "home.life_as_dev_5": "Kicked off career as a technical support engineer at 23",
        "home.life_as_dev_6": "Joined a startup as a junior software engineer at 24",
        "home.life_as_dev_7": "🆕Open for new opportunities!",
        "home.resume": "Resume",
        "home.whoami": "My name is Eric.",
        "navbar.about": "About",
        "navbar.home": "Home",
        "navbar.text": "Eric Lee",
        "work.conv": "Conversocial",
        "work.conv_jse": "Junior Software Engineer",
        "work.conv_work_period": "May 2019 – Apr 2020",
        "work.ibm": "IBM",
        "work.ibm_intern": "e-Commerce Java Development Support Intern",
        "work.ibm_work_period": "May 2016 – Apr 2018",
        "work.kingston_on": "Kingston, Ontario",
        "work.london_uk": "London, United Kingdom",
        "work.queensu": "Queen's University",
        "work.queensu_ta": "Teaching Assistant",
        "work.queensu_work_period": "Sep 2017 – Apr 2018",
        "work.toronto_ca": "Toronto, Canada",
        "work.verv": "Verv",
        "work.verv_tse": "Technical Support Engineer",
        "work.verv_work_period": "Oct 2018 – Apr 2019",
      }
    }
    onError={[Function]}
    textComponent={Symbol(react.fragment)}
  >
    <NavBar />
    <BrowserRouter>
      <Switch>
        <Route
          component={[Function]}
          path="/about"
        />
        <Route
          component={[Function]}
          exact={true}
          path="/"
        />
      </Switch>
    </BrowserRouter>
  </IntlProvider>
</div>
`;
